app/src/main/java/zapsolutions/zap/fragments/SendBSDFragment.java

```diff
@@ -504,85 +511,50 @@ public void onClick(View v) {
-                                        }
-                                    }, 300);
+                        if (mLnPaymentRequest.getNumSatoshis() <= RefConstants.LN_PAYMENT_FEE_THRESHOLD) {
+                            // ignore setting if below threshold
+                            if (mLnFeePercentCalculated >= 1f) {
+                                // fee higher than payment
+                                int feeSats = (int) (mLnPaymentRequest.getNumSatoshis() * mLnFeePercentCalculated);
+                                String feeLimitString = getString(R.string.fee_limit_exceeded_payment, feeSats, mLnPaymentRequest.getNumSatoshis());
+                                showFeeAlertDialog(srb, feeLimitString);
+                                return;
+                            } else {
+                                // could not calculate fee, or fee is below payment
+                                // set sanity fee, no one pays more fees than the value of payment
+                                srb.setFeeLimit(FeeLimit.newBuilder()
+                                        .setPercent(100)
+                                        .build());
+                            }
+                        } else {
+                            // check against fee setting
+                            if (mLnFeePercentSettingLimit != 1) {
+                                if (mLnFeePercentCalculated > mLnFeePercentSettingLimit) {
+                                    // fee is higher than settings, ask user
+                                    String feeLimitString = getString(R.string.fee_limit_exceeded, mLnFeePercentCalculated * 100, mLnFeePercentSettingLimit * 100);
+                                    showFeeAlertDialog(srb, feeLimitString);
+                                    return;
+                                } else {
+                                    // could not calculate fee, or fee is below limit
+                                    // set limit from settings and try payment
+                                    srb.setFeeLimit(FeeLimit.newBuilder()
+                                            .setPercent((long) (mLnFeePercentSettingLimit * 100)))
+                                            .build();
+                                }
+                            } else {
+                                // fee higher than payment
+                                if (mLnFeePercentCalculated >= 1f) {
+                                    int feeSats = (int) (mLnPaymentRequest.getNumSatoshis() * mLnFeePercentCalculated);
+                                    String feeLimitString = getString(R.string.fee_limit_exceeded_payment, feeSats, mLnPaymentRequest.getNumSatoshis());
+                                    showFeeAlertDialog(srb, feeLimitString);
+                                    return;
                                 }
```

Comment:
Is this reachable?

Reply:
`mLnFeePercentSettingLimit` will be 1 if the fee limit set to `None` (basically limit is 100%). Thats when we end up here and just do a "sanity check" if the fee would be higher than the payment as you mentioned in slack.